src/control/positionControl/BBTrajectories/WorldObjects.cpp

```diff
@@ -0,0 +1,183 @@
+            for (int i = 0; i < ballTrajectory.size(); i++) {
+                if (ruleset.minDistanceToBall > (pathPoints[i] - ballTrajectory[i]).length()
+                    || (gameState.getStrategyName() == "ball_placement_them"
+                        && ruleset.minDistanceToBall > ballTube.distanceToLine(pathPoints[i]))) {
+                    collisionDatas.emplace_back(CollisionData{ballTrajectory[i], pathPoints[i]});
+                    collisionTimes.emplace_back(i * timeStep);
+                    return;
+                }
+            }
+        }
+    }
+
+    void
+    WorldObjects::calculateEnemyRobotCollisions(rtt::BB::BBTrajectory2D BBTrajectory, std::vector<CollisionData> &collisionDatas,
+                                                std::vector<double> &collisionTimes, const std::vector<Vector2> &pathPoints,
+                                                double timeStep) {
+        auto theirRobots = world->getWorld()->getThem();
+
+        for (int j = 0; j < pathPoints.size(); j++) {
+            for (int i = 0; i < theirRobots.size() - 1; i++) {
+                double currentTime = j * timeStep;
+                double maxCollisionCheckTime = 2;
+                if (currentTime <= maxCollisionCheckTime) {
+                    // TODO: Currently enemy position in future is calculated with simple model (x = x + v*t), maybe improve?
+                    Vector2 theirVel = theirRobots[i]->getVel();
+                    Vector2 theirPos = theirRobots[i]->getPos() /*+ theirVel * currentTime*/;
+                    Vector2 posDif = BBTrajectory.getPosition(currentTime) -
+                                     theirPos; //TODO: Is currentTime defined the same way as the time from the BBTrajectory time?
+                    if (posDif.length() < 3 /*<-- kawaiii*/ * ai::Constants::ROBOT_RADIUS_MAX()) {
+                        Vector2 ourVel = BBTrajectory.getVelocity(currentTime);
+                        Vector2 velDif = ourVel - theirVel;
+                        double projectLength = velDif.dot(posDif) / sqrt(posDif.dot(posDif));
+                        if (abs(projectLength) > 1.5 && theirVel.length() < ourVel.length()) {
+                            collisionDatas.emplace_back(CollisionData{theirPos, pathPoints[j]});
+                            collisionTimes.emplace_back(currentTime);
+                            return;
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    void
+    WorldObjects::calculateOurRobotCollisions(std::vector<CollisionData> &collisionDatas, std::vector<double> &collisionTimes,
+                                              const std::vector<Vector2> &pathPoints, int robotId, double timeStep) {
+
+        int ourRobotAmount = world->getWorld()->getUs().size();
+
+        for (int j = 0; j < pathPoints.size(); j++) {
+            for (int i = 0; i < ourRobotAmount; i++) {
```

Comment:
Can be confusing, since people usually use i for the outer loop and j for the inner loop.